stm32-modules/heater-shaker/include/heater-shaker/host_comms_task.hpp

```diff
@@ -579,6 +579,31 @@ requires MessageQueue<QueueImpl<Message>, Message> class HostCommsTask {
         return std::make_pair(true, tx_into);
     }
 
+    template <typename InputIt, typename InputLimit>
+    requires std::forward_iterator<InputIt>&&
+        std::sized_sentinel_for<InputLimit, InputIt> auto
+        visit_gcode(const gcode::DebugControlPlateLockMotor& gcode,
+                    InputIt tx_into, InputLimit tx_limit)
+            -> std::pair<bool, InputIt> {
+        auto id = ack_only_cache.add(gcode);
+        if (id == 0) {
+            return std::make_pair(
+                false, errors::write_into(tx_into, tx_limit,
+                                          errors::ErrorCode::GCODE_CACHE_FULL));
+        }
+        auto message =
+            messages::SetPlateLockPowerMessage{.id = id, .power = gcode.power};
+        if (!task_registry->motor->get_message_queue().try_send(
```

Comment:
nit: there's quite a lot of duplicated  logic in the various `visit_gcode` functions.

Reply:
There really is, I don't like it. I think I can fix it (probably in a future PR) but it's going to be pretty templatey.